### ISSUE TITLE
[0.26.0-rc] autocorrect broken

### ISSUE THREAD

**Issue by sjmueller** [facebook/react-native#7496]
When setting up a `TextInput` to keep in sync with state (as per documentation), autocorrect breaks. For example:
```
<TextInput
  style={styles.title}
  placeholder="Title"
  onChangeText={title => this.setState({title})}
  value={this.state.title}
/>
```
It seems that the value of the native control is now getting overridden during every `onChangeText` event that updates state, because even things like double space no longer results in a period. 

![autocorrect](https://cloud.githubusercontent.com/assets/29085/15162798/bf3d5b5e-16b9-11e6-9563-d6c6495511e8.gif)

For reference, I'm also on `react 15.0.2`

**Comment by sjmueller** [comment]
Did some investigation, and found this commit:
##### Fix autocomplete in rich-text editing mode for CJK text input https://github.com/facebook/react-native/commit/91dcc9ac8e54ef4ba4592447cc484ac84d56c6b9

I rolled these changes out of my build, but still looks like the same problems persist. Any suggestions?

**Comment by mpretty-homepass** [comment]
@sjmueller It looks like the 'setText:' method of RCTTextField is getting called twice, once with the old value then again with the new value. Having it set twice is what is causing the problem.

**Comment by mpretty-homepass** [comment]
Actually, it looks like the problem is this if statement https://github.com/facebook/react-native/blob/dd8caf4cf21941aff5afc7e919ced58ec564f3a3/Libraries/Components/TextInput/TextInput.js#L619

**Comment by mpretty-homepass** [comment]
@nicklockwood It seems to me that autocorrect could never work if the TextInput is a controlled component. Unfortunately I can't think of a nice way to fix this without having something return that the text change is valid and hence shouldn't call setNativeProps. Do you have any ideas which might help?

**Comment by mpretty-homepass** [comment]
@sjmueller I've created a PR which seems to fix the autocorrect on iOS (haven't tested the effects on Android yet). If you decide to try it out then let me know if this introduces any bug.

**Comment by rigdern** [comment]
This is a regression caused by a change to batch event handling in React Native: https://github.com/facebook/react/commit/9f11f8c2634921e657df2691e6bfda18fead5bcc

**Comment by spicyj** [comment]
Not at a computer, but you can look at a recent change I did to PickerAndroid to fix a similar issue.

**Comment by rigdern** [comment]
@nicklockwood Should this issue be closed since you merged my PR? 26aa27da630a8875dfe64f616b6aca59eaa89aaf

**Comment by mpretty-homepass** [comment]
I can confirm the https://github.com/facebook/react-native/commit/26aa27da630a8875dfe64f616b6aca59eaa89aaf fix works for me as well.
